aya/src/programs/tc.rs

```diff
@@ -118,25 +164,41 @@ impl SchedClassifier {
         load_program(BPF_PROG_TYPE_SCHED_CLS, &mut self.data)
     }
 
-    /// Attaches the program to the given `interface` using the default options.
+    /// Attaches the program to the given `interface` using the TCX link
+    /// API in the first position by default, revert to the legacy netlink API otherwise.
+    /// For fine-grain control over link ordering use [`SchedClassifier::attach_with_options`].
     ///
     /// The returned value can be used to detach, see [SchedClassifier::detach].
     ///
     /// # Errors
     ///
-    /// [`TcError::NetlinkError`] is returned if attaching fails. A common cause
-    /// of failure is not having added the `clsact` qdisc to the given
+    /// [`TcError::NetlinkError`] is returned if attaching via netlink fails. A
+    /// common cause of failure is not having added the `clsact` qdisc to the given
     /// interface, see [`qdisc_add_clsact`]
     ///
     pub fn attach(
         &mut self,
         interface: &str,
         attach_type: TcAttachType,
     ) -> Result<SchedClassifierLinkId, ProgramError> {
-        self.attach_with_options(interface, attach_type, TcOptions::default())
+        if KernelVersion::current().unwrap() >= KernelVersion::new(6, 6, 0) {
```

Comment:
I'd be ok with fixing this as a follow up, but just ooc is there a way to probe
this at runtime?

Reply:
I don't think there is but I can try and determine one, in a follow up

Reply:
When TcAttachType is Custom, we should attach with netlink. If I've set it to
custom I know what I'm doing, I don't want you to try to use Tcx and fail

Reply:
Good catch.  This is a bug.  I've added a check to prevent this.